src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/PrefetchGetRecordsCache.java

```diff
@@ -132,7 +145,7 @@ public void run() {
                     } catch (InterruptedException e) {
                         log.info("Thread was interrupted, indicating shutdown was called on the cache.");
                         callShutdownOnStrategy();
-                    } catch (Error e) {
+                    } catch (Throwable e) {
```

Comment:
When running in synchronous this can catch exceptions thrown from the SDK.  These exceptions aren't serious errors, and shouldn't be caught here.  This is specifically the last ditch catch to prevent thread death.

Reply:
So should we not be logging the exceptions or should we try to catch the other kinds of exceptions and then the throwable?